### PR TITLE
fix(catalog-management): update examples after recent api changes

### PR DESCRIPTION
**Pull request by Andras-Csanyi**
## PR summary

Examples are aligned to the updated api.
Worth to consider merge this PR after this one: https://github.com/IBM/platform-services-go-sdk/pull/130

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by pyrooka — APPROVED** [review]
LGTM!

File: catalogmanagementv1/catalog_management_v1_examples_test.go [hunk 1]
```diff
@@ -0,0 +1,1728 @@
...
+)
+
+//
+// This file provides an example of how to use the Catalog Management service.
+//
+// The following configuration properties are assumed to be defined:
+// CATALOG_MANAGEMENT_URL=<service base url>
+// CATALOG_MANAGEMENT_AUTH_TYPE=iam
+// CATALOG_MANAGEMENT_APIKEY=<IAM apikey>
+// CATALOG_MANAGEMENT_AUTH_URL=<IAM token service base URL - omit this if using the production environment>
```
Comment (pyrooka) [inline catalogmanagementv1/catalog_management_v1_examples_test.go:39]: I skipped this accidentally. Could you update this section with all the variables currently in use?

File: catalogmanagementv1/catalog_management_v1_examples_test.go [hunk 2]
```diff
@@ -0,0 +1,1728 @@
...
+			}
+
+			// end-common
+
+			Expect(catalogManagementService).ToNot(BeNil())
+
+			token, err := catalogManagementService.Service.Options.Authenticator.(*core.IamAuthenticator).RequestToken()
+			Expect(err).To(BeNil())
+			bearerToken = token.RefreshToken
+			Expect(bearerToken).NotTo(BeNil())
```
Comment (pyrooka) [inline catalogmanagementv1/catalog_management_v1_examples_test.go:120]: Just an info here. There is a new Go core release with the `GetToken()` method (it's exported now). The go.mod has been already updated on the main branch so if you merge in that, you could rewrite this to:
```suggestion
			bearerToken, err := catalogManagementService.Service.Options.Authenticator.(*core.IamAuthenticator).GetToken()
			Expect(err).To(BeNil())
			Expect(bearerToken).NotTo(BeNil())
```

File: catalogmanagementv1/catalog_management_v1_examples_test.go (hunk 1 above)
Comment (Andras-Csanyi) [reply catalogmanagementv1/catalog_management_v1_examples_test.go:39]: Updated in all 4 SDK examples

File: catalogmanagementv1/catalog_management_v1_examples_test.go (hunk 2 above)
Comment (Andras-Csanyi) [reply catalogmanagementv1/catalog_management_v1_examples_test.go:120]: Thanks for the info! :)

File: catalogmanagementv1/catalog_management_v1_examples_test.go [hunk 3]
```diff
@@ -0,0 +1,1796 @@
...
+			shouldSkipTest()
+		})
+		It("Successfully construct the service client instance", func() {
+			var err error
+
+			// begin-common
+
+			catalogManagementServiceOptions := &catalogmanagementv1.CatalogManagementV1Options{}
+
+			catalogManagementService, err = catalogmanagementv1.NewCatalogManagementV1UsingExternalConfig(catalogManagementServiceOptions)
```
[the remark this answers is not in the document]
Comment (Andras-Csanyi) [reply catalogmanagementv1/catalog_management_v1_examples_test.go:107]: Nope, I started something and changed my mind and that part just left there. Nice catch!

**Pull request merged**